src/components/ResultsLayout.tsx

```diff
@@ -238,20 +241,42 @@ class ResultsLayout extends React.Component<Props, State> {
         this.setState({ notesModal: false })
     }
 
+    onActionPlanModalOpen = async (): Promise<boolean> => {
```

Comment:
I see what you mean with not needing the `async` there -- I think I was thinking I needed it with returning the promise, but I I don't -- only to resolve it. 

Now that I'm looking at this again, I don't think I need the `await` for `return await this.onActionPlanModal()` -- I Think we're only concerned with a single promise here, and whether it resolves to a boolean in `handleActionPlanModal` or the `preBack()` in the AppBar component, I don't think(...) it would cause a difference. (I think that's what you were getting at, but I have a lot more clarity looking at this right now, it's helping me to think through it.)

Which conditional are you referring to? the `this.state.awaitingConfirmationRef` in the ModalClose/Discard methods is just to make sure there's a value there, but the boolean passed to the `resolve` there do eventually make their way correctly to the AppBar component (l.266).

Gotta say, trying to solve this caused me a bit of a headache. Hopefully it's not too out there of a solution.

Reply:
Haha yeah, I get that. I think it's a solid solution. If you get the promises straightened out, I think it's in good shape!